x/distribution/client/cli/tx.go

```diff
@@ -40,6 +45,33 @@ func GetTxCmd(storeKey string, cdc *amino.Codec) *cobra.Command {
 	return distTxCmd
 }
 
+type generateOrBroadcastFunc func(context.CLIContext, authtxb.TxBuilder, []sdk.Msg) error
+
+func splitAndApply(
+	generateOrBroadcast generateOrBroadcastFunc,
+	cliCtx context.CLIContext,
+	txBldr authtxb.TxBuilder,
+	msgs []sdk.Msg,
+	chunkSize int,
+) error {
+	totalMessages := len(msgs)
+	if chunkSize == 0 {
+		chunkSize = totalMessages
```

Comment:
This function took me too long to grok, I suggest we:
- call `generateOrBroadcast` directly within this if statement (then return) -  there is no point in going through any of the chunk logic if we are broadcasting all the messages.
- include a simple comment above the for loop something like:
```
// split messages into arrays of length chunkSize then broadcast
```

Reply:
It was like that before but that makes unit testing very hard as it is impossible to decouple things

Reply:
That doesn't make sense... What I'm suggesting should not affect the outcomes of calling `splitAndApply` they should be identical to how they are now, this is just making the code more comprehensible

Reply:
Sorry, I misunderstood. Yes, I can call `generateOrBroadcast` directly there.